pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_show_trace failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_show_trace [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20236658?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20236658?buildTab=artifacts#/) on release-25.3 @ [4652ba096569ca15b4b64c472b6aeea4daa49e24](https://github.com/cockroachdb/cockroach/commits/4652ba096569ca15b4b64c472b6aeea4daa49e24):


```
                    ),
                    'mutationJobs:<[^>]*>',
                    'mutationJobs:<...>'
                  ),
                  e'drop_time:[1-9]\\d*',
                  'drop_time:...'
                ),
                'declarative_schema_changer_state:<[^>]*>',
                'declarative_schema_changer_state:<...>'
              ),
              'metadata:<[^>]*>',
              'metadata:<...>'
                 ) AS message
            FROM [SHOW KV TRACE FOR SESSION]
           WHERE message NOT LIKE '%Z/%'
                 AND operation NOT LIKE 'kv.DistSender: sending partial batch%'
                 AND message NOT SIMILAR TO '%(PushTxn|ResolveIntent|SystemConfigSpan)%'
                 AND tag NOT LIKE '%intExec=%'
                 AND tag NOT LIKE '%scExec%'
                 AND tag NOT LIKE '%IndexBackfiller%'
                 AND operation != 'dist sender send';
        
[01:42:07] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15799/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/show_trace: 30 statements
[01:42:08] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15799/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/show_trace with config local: 57 tests, 2 failures
[01:42:09] --- total progress: 57 statements
--- total: 57 tests, 2 failures
    logic.go:4549: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d9fd48d5b6db6734bd321fd2d9d3f428/logTestExecBuild_show_trace418092659
--- FAIL: TestExecBuild_show_trace (4.66s)
=== RUN   TestExecBuild_show_trace/system_table_lookup
    logic.go:3197: let $rangeid = 76
    logic.go:3147: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15799/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/show_trace:366: SELECT operation, message FROM [SHOW KV TRACE FOR SESSION]
        WHERE message     LIKE '%r76: sending batch%'
          AND message NOT LIKE '%PushTxn%'
          AND message NOT LIKE '%QueryTxn%'
        expected:
            dist sender send  r76: sending batch 4 CPut, 1 EndTxn to (n1,s1):1
            dist sender send  r76: sending batch 5 Get to (n1,s1):1
            dist sender send  r76: sending batch 1 EndTxn to (n1,s1):1
            
        but found (query options: "") :
            dist sender send  r76: sending batch 4 CPut, 1 EndTxn to (n1,s1):1
            dist sender send  r76: sending batch 5 CPut to (n1,s1):1
            dist sender send  r76: sending batch 1 EndTxn to (n1,s1):1
            
    logic.go:2335: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15799/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/show_trace:376: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_show_trace/system_table_lookup (0.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_show_trace.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52858

--- a/pkg/sql/opt/exec/execbuilder/testdata/show_trace
+++ b/pkg/sql/opt/exec/execbuilder/testdata/show_trace
@@ -5,6 +5,13 @@
 statement ok
 SET kv_transaction_buffered_writes_enabled = true
 
+# The buffer size can be changed metamoprhically, and if it happens to be too
+# small (around 1KiB), then some of the txns below might flush the buffer,
+# resulting in different KV requests. To ensure deterministic traces we always
+# update the buffer size to be large enough for txns in this file.
+statement ok
+SET CLUSTER SETTING kv.transaction.write_buffering.max_buffer_size = '2KiB';
+
 # Check SHOW KV TRACE FOR SESSION.
 
 let $trace_query

